snapshots.go

```diff
@@ -621,3 +622,262 @@ func (s *Stream) SnapshotToDirectory(ctx context.Context, dir string, opts ...Sn
 		return progress, nil
 	}
 }
+
+// SnapshotToBuffer creates a backup into s2 compressed tar file
+func (s *Stream) SnapshotToBuffer(ctx context.Context, dataBuffer, metadataBuffer io.Writer, opts ...SnapshotOption) (SnapshotProgress, error) {
```

Comment:
I've created this as a new function to avoid breaking interface, but would be open to other ideas to integrate this more gracefully.

Reply:
No I think this is good. But rather than duplicate most of the code I'd see if we could come up with a common set of code.

Imagine the logic for extracting is only in this one that takes the Writers then `RestoreSnapshotFromDirectory()` and `SnapshotToDirectory()` could conern themselves just with opening the files in the directory and passing those in here where the bulk of the logic lives.

As the PR is now this would be error prone if we ever need to update this logic - which is likely as we know we have some improvements in this area